library/core/class.dispatcher.php

```diff
@@ -796,6 +797,13 @@ private function _mapParts($Parts, $ControllerKey) {
             }
         }
     }
+
+    /**
+     * Trigger an event allowing addons to modify response headers.
+     */
+    public function sendHeaders() {
```

Comment:
FYI, this dispatcher change borked an older forum running on PHP 5.4. The signin/signout leads to a 500 error. I rewrote as:

` header_register_callback($this->fireEvent('SendHeaders'));`

And it worked! This is more just an FYI for those open sourcers trying to live on the edge. Anyway it's time to upgrade to a new version of PHP.  I am sure other stuff will start to fail in 5.4, so it's time to move on up :)